src/cascadia/TerminalSettingsModel/TerminalSettings.cpp

```diff
@@ -88,53 +96,100 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
     //     StartingDirectory) in this object to override the settings directly from
     //     the profile.
     // - keybindings: the keybinding handler
-    Model::TerminalSettings TerminalSettings::CreateWithNewTerminalArgs(const Model::CascadiaSettings& appSettings,
-                                                                        const Model::NewTerminalArgs& newTerminalArgs,
-                                                                        const IKeyBindings& keybindings)
+    // Return Value:
+    // - the GUID of the created profile, and a fully initialized TerminalSettings object
+    Model::TerminalSettingsStruct TerminalSettings::CreateWithNewTerminalArgs(const CascadiaSettings& appSettings,
+                                                                              const NewTerminalArgs& newTerminalArgs,
+                                                                              const IKeyBindings& keybindings)
     {
         const guid profileGuid = appSettings.GetProfileForArgs(newTerminalArgs);
-        auto settings{ CreateWithProfileByID(appSettings, profileGuid, keybindings) };
+        auto settings = CreateWithProfileByID(appSettings, profileGuid, keybindings);
 
         if (newTerminalArgs)
         {
             // Override commandline, starting directory if they exist in newTerminalArgs
             if (!newTerminalArgs.Commandline().empty())
             {
-                settings.Commandline(newTerminalArgs.Commandline());
+                settings.DefaultSettings().Commandline(newTerminalArgs.Commandline());
             }
             if (!newTerminalArgs.StartingDirectory().empty())
             {
-                settings.StartingDirectory(newTerminalArgs.StartingDirectory());
+                settings.DefaultSettings().StartingDirectory(newTerminalArgs.StartingDirectory());
             }
             if (!newTerminalArgs.TabTitle().empty())
             {
-                settings.StartingTitle(newTerminalArgs.TabTitle());
+                settings.DefaultSettings().StartingTitle(newTerminalArgs.TabTitle());
             }
             if (newTerminalArgs.TabColor())
             {
-                settings.StartingTabColor(static_cast<uint32_t>(til::color(newTerminalArgs.TabColor().Value())));
+                settings.DefaultSettings().StartingTabColor(static_cast<uint32_t>(til::color(newTerminalArgs.TabColor().Value())));
             }
             if (newTerminalArgs.SuppressApplicationTitle())
             {
-                settings.SuppressApplicationTitle(newTerminalArgs.SuppressApplicationTitle().Value());
+                settings.DefaultSettings().SuppressApplicationTitle(newTerminalArgs.SuppressApplicationTitle().Value());
             }
         }
 

```

Comment:
are there no appearance settings from a NewTerminalArgs?

Reply:
Not as of yet!

Reply:
I think new terminal args should only touch the default appearance for now